src/focus-camera/focus-first-residue.ts

```diff
@@ -60,12 +60,44 @@ export class FocusFirstResidue implements FocusFactoryI {
             imageRender.canvas3d.camera.setState(state);
         };
     }
-    getAxesToFlip(positions: Float32Array, origin: Vec3, up: Vec3, normalDir: Vec3) {
-        const toYAxis = calculateDisplacement(positions, origin, normalDir);
-        const toXAxis = calculateDisplacement(positions, origin, up);
+    getAxesToFlip(position: Vec3, origin: Vec3, up: Vec3, normalDir: Vec3) {
+        const toYAxis = calculateDisplacement(position, origin, normalDir);
+        const toXAxis = calculateDisplacement(position, origin, up);
         const Axes: string[] = [];
-        if (toYAxis[0] < 0) Axes.push('aroundY');
-        if (toXAxis[0] < 0) Axes.push('aroundX');
+        if (toYAxis < 0) Axes.push('aroundY');
+        if (toXAxis < 0) Axes.push('aroundX');
         return Axes;
     }
+}
+
+function getFirstResidueOrAveragePosition(structure: Structure, caPositions: Float32Array): Vec3 {
+    //TODO Is this the best way to test single chain?
+    // if only one chain => first residue coordinates
+    if (structure.units.length === 1) {
+        return Vec3.create(caPositions[0], caPositions[1], caPositions[2]);
+    } else {
+    // if more than one chain => average of coordinates of the first chain
+        const tmpMatrixPos = Vec3.zero();
+        const AtomIndexs = structure.units[0].elements;
+        const firstChainPositions = [];
+        for (let i = 0; i < AtomIndexs.length; i++) {
```

Comment:
Wouldn't it be better to average just the CA positions to be more consistent?

Reply:
The AtomIndexs are the CA index, so it is the average Ca position after all.

Reply:
No, `AtomIndexs = structure.units[0].elements` which are all the elements in the unit, not just the Ca.

Reply:
<img width="772" alt="Screenshot 2022-12-24 at 5 29 44 PM" src="https://user-images.githubusercontent.com/92659167/210029863-6e583776-4502-4916-b478-55aecbf28979.png">
In the above two cases, where we render two chain in both symmetric structure 1a6d and asymmetric 12e8 structure with specified chain with operator name if it is symmetric. The `AtomIndexs` is the Ca index.

<img width="705" alt="Screenshot 2022-12-29 at 7 20 16 PM" src="https://user-images.githubusercontent.com/92659167/210031067-33588cbc-2b29-4481-880b-f5d2505a4e06.png">
However, in the case of symmetric chain rendering without specified operator name like above. The original location `structure.units[0].elements` store all elements (first console log), but `structure.units[0].props.polymerElements` is storing Ca(second console log), in the other case, this is undefined; As a result, I change the code to the following to make sure the consistency. 

```let atomIndexs;
        if (structure.units[0].props.polymerElements) {
            atomIndexs = structure.units[0].props.polymerElements;
        } else {
            atomIndexs = structure.units[0].elements;
        }
```

Reply:
From which I found: 
Ca elements are in `structure.units[0].elements` if we call the chain specifically (chain name in asymmetric structure, chain name with operator in symmetric structure) and `structure.units[0].props.polymerElements` is undefined. 

If we only have chain name for symmetric structure, Ca elements is in `structure.units[0].props.polymerElements` and `structure.units[0].elements` will return every elements.

Is there a better way to access the Ca elements?